wp_api/src/api_error.rs

```diff
@@ -450,20 +451,22 @@ pub enum RequestExecutionErrorReason {
     },
     #[error(
         "The server at {} requires authentication. Please provide your username and password.",
-        url
+        hostname
     )]
     HttpAuthenticationRequiredError {
-        url: String,
+        hostname: String,
         server_message: Option<String>,
     },
     #[error(
         "The server at {} rejected your credentials. Please provide a valid username and password.",
-        url
+        hostname
     )]
     HttpAuthenticationRejectedError {
-        url: String,
-        server_message: Option<String>,
+        hostname: String,
+        method: Option<HttpAuthMethod>,
     },
+    #[error("The server is sending invalid HTTP authentication information. Please check your site's HTTP authentication configuration.")]
+    MisconfiguredHttpAuthenticationError { issue: HttpAuthMethodParsingError },
```

Comment:
Maybe we don't have to pre-parse the `www-authentication` header? Instead, we can include the header string in the error variant and expose an API to parse the string to `Result<HttpAuthMethod, HttpAuthMethodParsingError>`. By doing that, we can remove this `MisconfiguredHttpAuthenticationError` variant.

Reply:
Yeah, this is a weird one – the reason it exists is because we can't cover the entire world of HTTP Auth responses we might get. We could defer to the client for this, but we're modelling the rest of the `400` error space, so it seems odd to leave this one case out.